### DISCUSSION TITLE
Can we get support for uppercase DOCTYPE?

### DISCUSSION THREAD

**Discussion by bobbykjack** [prettier/prettier#16668] [Ideas]
[This issue](https://github.com/prettier/prettier/issues/15096) is closed, but I don't think it's been resolved. Parsing HTML with an XML parser is a perfectly valid use case which this transformation makes impossible.

At the very least, I think there should be an option so we don't have to run every HTML file through another process to convert "doctype" to "DOCTYPE".

**Comment by lulunac27a** [comment]
HTML DOCTYPE is case-sensitive, so both doctype and DOCTYPE works with no change in rendering HTML documents.

**Reply by bobbykjack** [reply]
Yes, but in XML—and, therefore, XHTML—it isn't and needs to be uppercase.
